pkg/kubelet/status/status_manager.go

```diff
@@ -435,6 +436,18 @@ func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
 				glog.V(3).Infof("Pod %q is terminated, but some containers are still running", format.Pod(pod))
 				return
 			}
+
+			// remove the safe-delete finalizer prior to no-grace deletion.
+			finalizers := sets.NewString(pod.GetFinalizers()...)
+			if finalizers.Has("k8s.io/safe-delete") {
+				finalizers.Delete("k8s.io/safe-delete")
+				pod.SetFinalizers(finalizers.List())
+				_, err := m.kubeClient.Core().Pods(pod.Namespace).Update(pod)
```

Comment:
I think it would still be safe I think. It would remove the finalizer and be ready for deletion by the node controller in the next sync loop. By this point, the containers have been killed, so, it is safe to delete the pod object from the apiserver any time in the future.